src/extension/options-page/DashboardDialogs/Wallet.tsx

```diff
@@ -345,9 +369,10 @@ export function DashboardWalletCreateDialog(props: WrappedDialogProps<object>) {
                         variant="contained"
                         onClick={onSubmit}
                         disabled={
-                            !(state[0] === 0 && name && confirmed) &&
-                            !(state[0] === 1 && name && mnemonic) &&
-                            !(state[0] === 2 && name && privKey)
+                            (!(state[0] === 0 && name && confirmed) &&
+                                !(state[0] === 1 && name && mnemonic) &&
+                                !(state[0] === 2 && name && privKey)) ||
+                            name.length > 32
```

Comment:
```suggestion
                            name.length > 32
```

extract to a function. (e.q: `isAvailableWalletName`)